01_decouverte_micropython/01_decouverte_micropython.md

```diff
@@ -299,6 +299,75 @@ Modifie ta barre de chargement pour qu'au lieu d'allumer les diodes d'un coup
 (avec une intensité de 9), elles s'allument en douceur (en faisant
 progressivement varier l'intensité de 0 à 9).
 
+## L'indentation
+
+À partir de maintenant, tes programmes commenceront à être plus compliqués. Il
+est tout de même nécessaire de comprendre une partie importante de la syntaxe
+de Python : l'**indentation**.
+
+L'indentation correspond au nombre d'espaces avant le début d'une ligne. Par
+exemple, la première ligne ci-dessous à une indentation égale à **0** (pas
+d'espaces en début de ligne), et la deuxième ligne à une indentation égale à
+**4** (quatre espaces en début de ligne).
+
+```python
+for i in range(5):
+    display.scroll(str(i))
+```
+
+Cette indentation permet à Python de savoir, par exemple, quelles parties d'un
+programme sont à répéter. Comparons deux exemples.
+
+Dans l'exemple ci-dessous, la série suivante de chaînes de caractères est
+affichée :
+- 0
+- end
+- 1
+- end
+
+```python
+for i in range(2):
```

Comment:
Je n'ai pas regardé le reste du TP, est-ce qu'on se limite à un type de boucle (for ou while) ?

Reply:
On voit `for` juste au dessus et `while` juste en dessous. Je voulais introduire la notion d'intention le plus tôt possible, mais avant `for` je trouvais ça bizarre car je n'avais aucun exemple a donner, et après `while` je trouve que c'est trop tard. Et comme on a pas encore vu `while` a ce point j'utilise uniquement l'exemple de `for`.

Reply:
Ça me va, je vérifiais juste :)